From 760132d60486df3de090fd3863bde51a47e4f250 Mon Sep 17 00:00:00 2001
From: Simon Horman <horms+renesas@verge.net.au>
Date: Wed, 3 Jul 2013 13:38:20 +0900
Subject: [PATCH 0941/1083] Revert "drm/rcar-du: Don't clean up mode setting if
 it hasn't been initialized"

This reverts commit 6ea8d8cfd67389f3c49592484504ca022ec92529.
(cherry picked from commit 3c03cf9d1227405fba61771b8adcd546a0f73b1a)

Signed-off-by: Ryo Kataoka <ryo.kataoka.wt@renesas.com>
---
 drivers/gpu/drm/rcar-du/rcar_du_drv.c | 12 ++++++++----
 1 file changed, 8 insertions(+), 4 deletions(-)

diff --git a/drivers/gpu/drm/rcar-du/rcar_du_drv.c b/drivers/gpu/drm/rcar-du/rcar_du_drv.c
index 87ac205..813d48c 100644
--- a/drivers/gpu/drm/rcar-du/rcar_du_drv.c
+++ b/drivers/gpu/drm/rcar-du/rcar_du_drv.c
@@ -136,27 +136,31 @@ static int rcar_du_load(struct drm_device *dev, unsigned long flags)
 	mem = platform_get_resource(pdev, IORESOURCE_MEM, 0);
 	if (mem == NULL) {
 		dev_err(&pdev->dev, "failed to get memory resource\n");
-		return -EINVAL;
+		ret = -EINVAL;
+		goto done;
 	}
 
 	ioarea = devm_request_mem_region(&pdev->dev, mem->start,
 					 resource_size(mem), pdev->name);
 	if (ioarea == NULL) {
 		dev_err(&pdev->dev, "failed to request memory region\n");
-		return -EBUSY;
+		ret = -EBUSY;
+		goto done;
 	}
 
 	rcdu->mmio = devm_ioremap_nocache(&pdev->dev, ioarea->start,
 					  resource_size(ioarea));
 	if (rcdu->mmio == NULL) {
 		dev_err(&pdev->dev, "failed to remap memory resource\n");
-		return -ENOMEM;
+		ret = -ENOMEM;
+		goto done;
 	}
 
 	rcdu->clock = devm_clk_get(&pdev->dev, NULL);
 	if (IS_ERR(rcdu->clock)) {
 		dev_err(&pdev->dev, "failed to get clock\n");
-		return -ENOENT;
+		ret = -ENOENT;
+		goto done;
 	}
 
 	/* DRM/KMS objects */
-- 
1.8.3.2

